scripts/tests/core-integration.test.mjs

```diff
@@ -0,0 +1,447 @@
+import { describe, expect, it, beforeAll, afterAll } from "vitest";
+import * as fs from "node:fs/promises";
+import * as path from "node:path";
+import { fileURLToPath } from "node:url";
+import { glob } from "glob";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+describe("CORE Integration Tests", () => {
+  const testDir = path.join(__dirname, ".core-test-tmp");
+  const coreRepo = path.join(testDir, "pepr");
+  const docsDir = path.join(coreRepo, "docs");
+  const outputDir = path.join(testDir, "output");
+
+  beforeAll(async () => {
+    await fs.mkdir(testDir, { recursive: true });
+    await fs.mkdir(coreRepo, { recursive: true });
+    await fs.mkdir(docsDir, { recursive: true });
+    await fs.mkdir(outputDir, { recursive: true });
+  });
+
+  afterAll(async () => {
+    try {
+      await fs.rm(testDir, { recursive: true, force: true });
+    } catch (error) {
+      console.warn("Failed to clean up test directory:", error);
+    }
+  });
+
+  describe("Documentation Directory Discovery", () => {
+    it("should find all .md files in docs directory", async () => {
+      const testFiles = [
+        "010_user-guide/getting-started.md",
+        "010_user-guide/README.md",
+        "020_actions/mutate.md",
+        "README.md",
+        "index.js",
+      ];
+
+      for (const file of testFiles) {
+        const filePath = path.join(docsDir, file);
+        await fs.mkdir(path.dirname(filePath), { recursive: true });
+        await fs.writeFile(filePath, "# Test\n\nContent", "utf8");
+      }
+
+      const foundFiles = await glob(`${docsDir}/**/*.md`);
+      const relativePaths = foundFiles.map(f => path.relative(docsDir, f));
+
+      expect(relativePaths).toContain("010_user-guide/getting-started.md");
+      expect(relativePaths).toContain("010_user-guide/README.md");
+      expect(relativePaths).toContain("020_actions/mutate.md");
+    });
+
+    it("should exclude non-.md files", async () => {
+      const foundFiles = await glob(`${docsDir}/**/*.md`);
+      const foundBasenames = foundFiles.map(f => path.basename(f));
+
+      expect(foundBasenames).not.toContain("index.js");
+    });
+
+    it("should filter out root README.md from processing", async () => {
+      const files = ["README.md", "010_user-guide/README.md", "getting-started.md"];
+
+      // Simulate the filtering logic from findSourceDocFiles
+      const filtered = files.filter(f => f.endsWith(".md")).filter(f => !(f === "README.md"));
+
+      expect(filtered).not.toContain("README.md");
+      expect(filtered).toContain("010_user-guide/README.md");
+      expect(filtered).toContain("getting-started.md");
+    });
```

Comment:
This seems like testing an implementation detail of what `findSourceDocFiles` is supposed to do. Can we instead have a test around the results of when `findSourceDocFiles` works either as expected or under an unexpected situation?